### ISSUE TITLE
Roth - Blender 2.81 dae export

### ISSUE THREAD

**Issue by AdaRadius** [RuthAndRoth/Ruth#33]
I don't know who is reading which forums - I posted this yesterday in a couple of them. Mostly to see if I can find kindred souls who understand what I'm trying to do.

I pulled back from Roth testing, to test Blender 2.81 collada export itself, after much futile searching for anyone who has any anything other than cussing and "doesn't work" to say about it. Finally got a test working yesterday - only with the basic bones so far. Not volume bones nor the fingers, which I'll tackle today or tomorrow depending on how RL goes. 

![CubelyTest2Inworld_001](https://user-images.githubusercontent.com/28539799/71684688-fe918180-2d4a-11ea-9b75-429c75a32fee.png)

Meet MsCubelyV1.0, snapped inworld at the RuthAndRoth grid, with armature in the display and full alpha on the default avatar. I'll go in now and make sure there's a model out and that it's the right one, correctly labeled. :)  Then it's RL for me for a while, which gives me time to brood on it some more.

**Comment by AdaRadius** [comment]
This model seems to work with the Appearance sliders, although only the sliders that use the basic bone set.  I've only tried it with Firestorm so far and on this grid.

**Comment by aiaustin** [comment]
I saw all the Discord channels, RuthAndRoth (preferred) and Avastar (too many separate named threads to follow it all)... as well as here on GitHub if course.

**Comment by aiaustin** [comment]
The copy of MsCubely 
on OSGrid RuthAndRoth region seems to work fine Ada.  What is the tiny white bone that is just to the right of the centre line near the middle of the chest?
![2020-02-02-MsCubely-Bones](https://user-images.githubusercontent.com/4630885/71687394-a6cc3800-2d95-11ea-8274-0aaafb9c5a8a.jpg)

**Comment by AdaRadius** [comment]
That is mChest, inexplicably small. I don't know why the 2005 LL artists drew it that size, or at that angle.  Their design of mCollar is another mystery - most armatures have it perpendicular, attached to the upper arm bone.

**Comment by aiaustin** [comment]
Its also set 5 or 6cm to the right of the centre line. Odd.

**Comment by AdaRadius** [comment]
It shouldn't be. I'll go check that. Thank you!!

**Comment by AdaRadius** [comment]
The center bits of my mesh are a little out of alignment, .000178m for mChest, which I'm fixing. Puzzling.  No, not really, I crashed the software a couple of times trying to add the volume bones - and I might not have rechecked everything. This copy of the armature is aligned, whew. I'll upload a new version, then go back to figuring out what I can do about the next set of bones we need. Thank you!

**Comment by aiaustin** [comment]
I am not seeing a tiny 0.000178m offset Ada.  I saw MsCubely’s little mChest bone as 0.05m (5cm) or more from the centre line. For the copy on your test area on OSGrid RuthAndRoth region. I will go back and recheck when you indicate a new version is in place.

**Comment by AdaRadius** [comment]
If you were in a pose that shifts the bone, it would be moved away from center - I think all of the default stands move mChest. I realized that we don't have enough pose stands and other animation tools set up at my dock; I decided to work on that today. Which means gathering the thousands of animations I have in my Kitely avatar, sorting them into something useful, and handing them off to my OSGrid avatar. One of the stands will have the symmetrical poses - T-Pose, A-Pose for testing symmetry. 

I deleted yesterday's model, and put out the one with the corrections this morning. The mesh bits don't have to exactly reflect the armature to use for testing, but they should at least center on them. :)

**Comment by Outworldz** [comment]
There is no need to upload top check weight, except as a final and for 
tweaks. Blender plays BVH and AMNIM

  A small clarification:  There are no bones (except the tongue) that 
can "shift".   They rotate only.    The Bento tongue is the only one 
that can actually move so it can can protrude the tongue.   But it's a 
complicated subject due to the 110 bones in the armature, and in total 
392 used for many other purposes. There is a lot of bone wonkiness in 
Blender as it is not 3DSMax/Maya  which is what the original avatar was 
made with, E.g. the Pelvis Inverted bone.

if you want a collection of the standard SL poses, they are in this link 
under "Default Animations".  This is the "well known" set, which 
includes the facial emotes for the standard avatar's shape keys.  The 
other really useful set is in the SecondLife-POSER-BVH folder.

https://www.dropbox.com/sh/49nehmo63101ewg/AAAqTxTYbMQYXSAywOWecrW_a?dl=1

You can play any animation on any armature in Blender with these. No 
need to upload to check things.    If anyone needs instructions on 
uploading a BVH to blender and attaching it to an armature, please let 
me know and I will document what I do.  It's not hard, but is a tad 
tricky.  I do it all the time to check weights.

Short Blender video used to check a walk and chest movement for Debbie's 
mesh troll: http://www.outworldz.com/video/tezzi.mp4

Fred

**Comment by aiaustin** [comment]
> The Bento tongue is the only one that can actually
> move so it can can protrude the tongue.

That might explain why our included bento head tongue sticks through the lower lip in animations... whereas the heads that can be purchased in SL don’t seem to have that issue.

**Comment by AdaRadius** [comment]
@Outworldz Have you uploaded a model rigged in Blender 2.81 that has volume and Bento (hand) weights? That's why I'm uploading the Cubely model, to figure out what's borking on something simple, not to check weights themselves. So far I'll I've got working is the basic bone set.

**Comment by aiaustin** [comment]
@AdaRadius I checked out the replaced MsCubely on OSGrid RuthAndRoth region.. the little centred mChest bone now looks to me to be in the right place on the in world static model,.  I could not take a copy to test when worn... its "no copy" at the moment.  See image here...

![Centre-Bone-Okay-But-Not-Take-Copy](https://user-images.githubusercontent.com/4630885/71768558-ebc7aa00-2f0e-11ea-9fc2-375f2ec72e8e.jpg)

If you are happy with your DEFAULT being full perms for uploaded stuff, as I do, and to make sure all components, textures, scripts, etc when created or uploaded are full perm.. you can set the defaults via (in Firestorm) Build -> Options -> Set Default permissions.  I tick everything there.

![Firestorm-Default-Permissons-All-On](https://user-images.githubusercontent.com/4630885/71768614-76a8a480-2f0f-11ea-8005-adec24bd451f.jpg)

**Comment by AdaRadius** [comment]
No, I'll go fix the perms, I forgot about slambit - I am importing full perm (though I should go check it again)  and sometimes it doesn't work. And I'll give you edit rights on my stuff. 
I have set the animation pose gizmos set so that only the empty one is copiable because some of the animation artists don't want their stuff distributed in batches like that. They're sorted by AO types, plus three more, and as I work with them I'll be able to pare them down to simpler sets. I am interested in being able to test for clipping and good joint deformation when I figure out the export issue and get back to Roth.

**Comment by AdaRadius** [comment]
Fixed. Sorry!
For dance animations for testing Ruth and Roth, we maybe should consider using one or more of the Paramour scripted tools - the sim owner or administrator has to set those up, as far as I know.  I have most of Aine's dance animations I think, though I haven't gone to her grid in a while.

**Comment by aiaustin** [comment]
OSGrid RuthAndRoth region should now have the settings that allow the paramour dance globe controller to work correctly. I placed a dance controller over your Testing area.  If you want to place your own just delete the one I placed... or ask me to if it won't let you.

![2020-01-04-OSGrid-RuthAndRoth-Paramour-Dances-Single](https://user-images.githubusercontent.com/4630885/71771513-11b37580-2f34-11ea-852c-a40f4af2dd79.jpg)
![S2020-01-04-OSGrid-RuthAndRoth-Paramour-Dances-NPC-Couple-2](https://user-images.githubusercontent.com/4630885/71771502-e03aaa00-2f33-11ea-947b-4798e48c1b27.jpg)
![2020-01-04-OSGrid-RuthAndRoth-Paramour-Dances-NPC-Couple-1](https://user-images.githubusercontent.com/4630885/71771506-fcd6e200-2f33-11ea-9864-3b1fb14ff300.jpg)

**Comment by AdaRadius** [comment]
WOOT that's perfect. Thank you! I'll go look after I have a draft of Cubely's hands to bring in. I keep crashing Blender while trying to export after adding to the basic bones, so it's taking longer than I think it should to figure out a workflow. No wonder I was getting frustrated with Roth.

**Comment by AdaRadius** [comment]
Cubely with hands is working - I did expect that, so now on to the bit that hasn't worked yet, the volume bones.
